exploration/2025/gf_reanalysis_fs.md

```diff
@@ -0,0 +1,433 @@
+---
+jupyter:
+  jupytext:
+    formats: ipynb,md
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.16.1
+  kernelspec:
+    display_name: ds-aa-nga-flooding
+    language: python
+    name: ds-aa-nga-flooding
+---
+
+# GloFAS reanalysis / Floodscan
+<!-- markdownlint-disable MD013 -->
+Validation of GloFAS reanalysis using Floodscan in relevant LGAs or pixels
+
+```python
+%load_ext jupyter_black
+%load_ext autoreload
+%autoreload 2
+```
+
+```python
+import pandas as pd
+import ocha_stratus as stratus
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+import matplotlib.cm as cm
+import matplotlib.colors as mcolors
+
+from src.datasources import glofas, codab
+from src.constants import *
+from src.utils import blob
+```
+
+## Load data
+
+```python
+station_name = "wuroboki"
+```
+
+```python
+df_ra = glofas.load_glofas_reanalysis(station_name=station_name)
+```
+
+```python
+df_ra
+```
+
+```python
+adm2 = codab.load_codab_from_blob(admin_level=2, aoi_only=True)
+```
+
+```python
+pcodes = BENUE_ADM2_PCODES
+```
+
+```python
+query = f"""
+SELECT *
+FROM public.floodscan
+WHERE pcode IN {tuple(pcodes)}
+"""
+```
+
+```python
+df_fs = pd.read_sql(
+    query, stratus.get_engine(stage="prod"), parse_dates=["valid_date"]
+)
+```
+
+```python
+df_sfed = df_fs[df_fs["band"] == "SFED"]
+```
+
+```python
+blob_name = f"{blob.PROJECT_PREFIX}/processed/floodscan/fs_benue_pixels_1998_2024.parquet"
+df_fs_pixels = stratus.load_parquet_from_blob(blob_name)
+```
+
+```python
+df_fs_pixels["pixel_id"] = (
+    df_fs_pixels["x"].astype(str) + "_" + df_fs_pixels["y"].astype(str)
+)
+```
+
+```python
+df_fs_pixels
+```
+
+## Compare pixels
+
+See which pixels have the best correlation with GloFAS reanalysis
+
+```python
+df_compare_pixels = df_ra.rename(columns={"time": "date"}).merge(df_fs_pixels)
+```
+
+```python
+dicts = []
+for (x, y), group in df_compare_pixels.groupby(["x", "y"]):
+    dicts.append(
+        {
+            "x": x,
+            "y": y,
+            "corr": group.corr(numeric_only=True).loc["dis24", "SFED"],
```

Comment:
Curious... what are your opinions on Pearson vs Spearman correlations here? I guess we could assume this to be a linear correlation so the default Pearson would be good, but I always think of Spearman being better for outliers. Not totally sure regardless